development/source-code-documentation.md

```diff
@@ -0,0 +1,187 @@
+# Documentation standards
+
+Code should be documented according to [JSDoc](https://jsdoc.app/) standards. Documentation comments should enable the
+TypeScript compatible [type checking](https://www.typescriptlang.org/docs/handbook/type-checking-javascript-files.html),
+if such type checking is supported by an editor or an IDE. For this we should follow the 
+[TypeScript JSDoc guidelines](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html).
+
+Documentation should follow the [ESLint JSDoc rules](https://eslint.org/docs/rules/valid-jsdoc).
+
+## Module
+An ES6 module should include a module name at the first line of the module file:
+```
+/** @module moduleName */
+```
+
+A module name should start from a lowercase letter in order to distinguish it from the class name.
+
+## Enums
+
+Define enums as:
+```
+/**
+ * Enum description
+ *
+ * @enum {string} */
+export const EnumType = {
+  VALUE_ONE: 'one',
+  VALUE_TWO: 'two'
+}
+```
+
+References:
+[TypeScript enum documentation guidelines](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#enum)
+
+
+## Classes
+
+```
+class C {
+  /**
+   * @param {number} data
+   */
+  constructor(data) {
+    // Property types can be inferred
+    this.name = "foo"
+
+    // or set explicitly
+    /** @type {string | null} */
+    this.title = null
+
+    // or simply annotated, if they're set elsewhere
+    /** @type {number} */
+    this.size
+    
+    // Properties can be declared as private
+    /** 
+     * @type {string}
+     * @private 
+     */
+    this._privateProp
+    
+    // Type info can be combined with description
+    /**
+     * A prop's description.
+     *
+     * @type {number}
+     */
+    this.propOne = 0
+    
+    // Description, type info, and visibility can be used together
+    /**
+     * A prop's description.
+     *
+     * @type {string}
+     * @private 
+     */
+    this._privatePropTwo = 0
```

Comment:
it seems like this should be the recommendation?

Reply:
I think so. Does it make sense? If so, I can add a note that this is the recommended approach.

Reply:
I think yes.

Reply:
OK, I'll mark this as the recommended one.